### ISSUE TITLE
Provide future-proof record constructors

### ISSUE THREAD

**Issue by dsheets** [mirage/ocaml-github#168]
Either default records (less control) or constructor functions (more control) would allow users to write future-proofed code that creates GitHub record types. This would let us move the GitHub record types off the breaking change list. See OCamlPro/opam-publish#45 for the original suggestion by @AltGr.

**Comment by kit-ty-kate** [comment]
I agree. I encountered the same issue while reviewing #280. However, i've tried `atdgen -v` but it seems like this option might have an issue https://github.com/ahrefs/atd/issues/433, so i'm hoping to be able to provide a new `Github_v` module which would provide future-proof constructors, once this is fixed.
